Library/Homebrew/cmd/audit.rb

```diff
@@ -23,65 +26,51 @@
 require "date"
 
 module Homebrew
+  RUBY_2_OR_LATER = RUBY_VERSION.split(".").first.to_i >= 2
+
   def audit
     formula_count = 0
     problem_count = 0
 
     strict = ARGV.include? "--strict"
-    if strict && ARGV.resolved_formulae.any? && RUBY_VERSION.split(".").first.to_i >= 2
-      require "cmd/style"
-      ohai "brew style #{ARGV.resolved_formulae.join " "}"
-      style
-    end
-
+    style = strict && ARGV.resolved_formulae.any? && RUBY_2_OR_LATER
     online = ARGV.include? "--online"
 
     ENV.activate_extensions!
     ENV.setup_build_environment
 
-    if ARGV.switch? "D"
-      FormulaAuditor.module_eval do
-        instance_methods.grep(/audit_/).map do |name|
-          method = instance_method(name)
-          define_method(name) do |*args, &block|
-            begin
-              time = Time.now
-              method.bind(self).call(*args, &block)
-            ensure
-              $times[name] ||= 0
-              $times[name] += Time.now - time
-            end
-          end
-        end
-      end
-
-      $times = {}
-      at_exit { puts $times.sort_by { |_k, v| v }.map { |k, v| "#{k}: #{v}" } }
-    end
-
-    ff = if ARGV.named.empty?
-      Formula
+    if ARGV.named.empty?
+      ff = Formula
+      files = [HOMEBREW_LIBRARY]
```

Comment:
I'm not sure why we'd want to scan all of `Library` (including the code in `Library/Homebrew`) in this case, as this command is only about auditing formulae. I'd make that just `files = ff.map(&:path)` or if that creates a too long argument list for invoking `rubocop`, make that `Tap.map(&:formula_dir)` instead (array of formula directories of all installed taps).

Here's my proposal for this line and some of the surrounding code:

```ruby
ff = if ARGV.named.empty?
  Formula
else
  ARGV.resolved_formulae
end

if style
  # Check style in a single batch run up front for performance
  files = ff.map(&:path)
  style_results = check_style_json(files, :realpath => true)
end
```